pkg/cmd/cli/cmd/expose.go

```diff
@@ -119,8 +120,15 @@ func validate(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
 			supportsTCP := false
 			for _, port := range svc.Spec.Ports {
 				if port.Protocol == kapi.ProtocolTCP {
-					// Pass service target port as the route port
-					cmd.Flags().Set("port", port.TargetPort.String())
+					// Pass endpoint port as the route port
+					endpoints, err := kc.Endpoints(svc.Namespace).Get(svc.Name)
+					if err != nil {
```

Comment:
You can't fail here - endpoints might not exist yet (because no pods).  Why aren't you looking at the spec.ports.targetPort field and applying the same logic?